File: tests/test_cpg/fixtures/parity_sqli.js
/** Cross-language SQL injection fixture — JavaScript (Express). */

var req = { query: {}, body: {} };
var db = { query: function() {}, execute: function() {} };

function search() {
    var keyword = req.query.q;  // $ source=sql_injection
    db.query("SELECT * FROM posts WHERE title LIKE '%" + keyword + "%'");  // $ sink=sql_injection
}

function getUser() {
    var uid = req.query.uid;  // $ source=sql_injection
    db.execute("SELECT * FROM users WHERE id = " + uid);  // $ sink=sql_injection
}

File: tests/test_cpg/fixtures/parity_sqli.py
"""Cross-language SQL injection fixture — Python (Flask)."""

from flask import Flask, request

app = Flask(__name__)


@app.route("/search")
def search():
    keyword = request.args.get("q", "")  # $ source=sql_injection
    cursor.execute(
        "SELECT * FROM posts WHERE title LIKE '%" + keyword + "%'"
    )  # $ sink=sql_injection


@app.route("/user/<int:user_id>")
def get_user(user_id):
    uid = request.args.get("uid")  # $ source=sql_injection
    cursor.execute(f"SELECT * FROM users WHERE id = {uid}")  # $ sink=sql_injection
